### ISSUE TITLE
Conditional remove a map - JOLT spec plz

### ISSUE THREAD

**Issue by yarabsyp** [bazaarvoice/jolt#553]
Input 
Need a spec for conditional removing maps in the json
```
{
  "phoneDetails": {
    "homePhone": {
      "pType": "H",
      "region": "U",
      "number": "1234567890",
      "transactionCode": "I"
    },
    "businessPhone": {
      "pType": "B"
    },
    "cellularPhone": {
      "pType": "C"
    },
    "additionalPhone": {
      "region": "U",
      "number": "1234567890",
    }
  },
  "ParticipantDetails": {
    "participantMethodValidation": "CDCR",
    "participantId": [
      {
        "idType": "CORP",
        "verificationText": "test"
      },
      {
        "idType": "CORP",
        "verificationText": "test"
      }
    ]
  }
}
```
1. I want to remove a map based on the key doesn't exist, Ex: "businessPhone" don't have a key region and number so i want to remove that ,as well as cellularPhone .
2. participantMethodValidation want to move this key into list of map 

expected output
```
{
  "phoneDetails": {
    "homePhone": {
      "pType": "H",
      "region": "U",
      "number": "1234567890",
      "transactionCode": "I"
    },
    "additionalPhone": {
      "region": "U",
      "number": "1234567890",
    }
  },
  "ParticipantDetails": {
    "participantId": [
      {
        "participantMethodValidation": "CDCR",
        "idType": "CORP",
        "verificationText": "test"
      },
      {
        "participantMethodValidation": "CDCR",
        "idType": "CORP",
        "verificationText": "test"
      }
    ]
  }
}
```

**Comment by milosimpson** [comment]
"I want to remove a map based on the key doesn't exist, Ex: "businessPhone" don't have a key region and number so i want to remove that ,as well as cellularPhone ."

The trick is to first use "default" to make there be known default values for "businessPhone" and "cellularPhone".

Second, match on those values and don't pass them thru.   Similar to https://github.com/bazaarvoice/jolt/issues/507 which you asked.    See also https://github.com/bazaarvoice/jolt/issues/557  




"participantMethodValidation want to move this key into list of map"

See https://github.com/bazaarvoice/jolt/issues/555  

Basically, in "shift" match into each element of the participantId array, and then use "@" to reference the "participantMethodValidation" to copy it to the output.
